### ISSUE TITLE
Objective-C call graph is not shown when self proper is called

### ISSUE THREAD

**Issue by samuelchen2015** [doxygen/doxygen#11102]
**Describe the bug**

Objective-C code is below,  `[self.eventHandler setup]; //  this call graph is not recongnized`。 I can see 
```
<ObjCCall,ObjCMName,ObjCSkipStr>.  {
                                      printf("<ObjCCall,ObjCMName,ObjCSkipStr>.\n");
                                      g_currentCtx->format+=*yytext;
                                      }
```
 in code.l, does it mean "." is ignore?

would you enhance this for OC？it is ok for java

**Screenshots**
If useful, add screenshots to help explain your problem.

**To Reproduce**
```
<ObjCCall,ObjCMName,ObjCSkipStr>.  {
                                      printf("<ObjCCall,ObjCMName,ObjCSkipStr>.\n");
                                      g_currentCtx->format+=*yytext;
                                      }
```

```
// ViewController.h
#import <UIKit/UIKit.h>
#import "EventHandler.h"

@interface ViewController : UIViewController

@property (nonatomic, strong) EventHandler *eventHandler;

- (instancetype)initWithEventHandler:(EventHandler *)eventHandler;

@end
```

```
// ViewController.m
#import "ViewController.h"
#import "ViewModel.h"

@implementation ViewController

- (instancetype)initWithEventHandler:(EventHandler *)eventHandler {
    self = [super initWithNibName:nil bundle:nil];
    if (self) {
        _eventHandler = eventHandler;
        [self.eventHandler setup]; //  this call graph is not recongnized
    }
    return self;
}
@end
```
**Expected behavior**
Describe what you would have expected or think is correct.

**Version**
Mention the version of doxygen used (output of `doxygen --version`) and the platform on which you run doxygen (e.g. Windows 10, 64 bit). If you run doxygen under Linux please also mention the name and version of the distribution used (output of `lsb_release -a`) and mention if you compiled doxygen yourself or that you use a binary that comes with the distribution or from the doxygen website.

**Stack trace**
If you encounter a crash and can build doxygen from sources yourself with debug info (`-DCMAKE_BUILD_TYPE=Debug`), a stack trace can be very helpful (especially if it is not possible to capture the problem in a small example that can be shared).

**Additional context**

**Comment by samuelchen2015** [comment]
thanks for reply

**Comment by samuelchen2015** [comment]
thanks for reply
doxygen version 1.8.17, yes a bit old version。I make some code develop base on this version 

doxyconfig, example code  is attach. 
[doxygen.zip](https://github.com/user-attachments/files/16743992/doxygen.zip)

I wonder whether "." is consider in the following code of code.l 

<ObjCCall,ObjCMName>"["|"{"       {
                                    saveObjCContext();
			            g_currentCtx->format+=*yytext;
			            BEGIN(ObjCCall);
		                    //printf("open\n");
                                  }
<ObjCCall,ObjCMName>"]"|"}"       {
			            g_currentCtx->format+=*yytext;
                                    restoreObjCContext();
			            BEGIN(ObjCMName);
		             	    if (g_currentCtx==0)
				    {
				      // end of call
				      writeObjCMethodCall(g_contextDict.find(0));
				      BEGIN(Body);
				    }

**Comment by albert-github** [comment]
Doxygen version 1.8.17 is quite old (December 27, 2019) so I definitely would advise to update to the current 1.12.0 doxygen version.

You got the warning:
```
error: file name expected for option GENERATE_TAGFILE, got NO instead. Ignoring...
```
`GENERATE_TAGFILE` is  not a boolean value but needs a filename and when left blank no tag file is generated.

I don't get the:
> I wonder whether "." is consider in the following code of code.l

- with "." do you mean the wildcard, any character, dot or a literal dot (which would normally show as `\.` in the rule though)
- The given rules are regarding open and closing brackets.
- I'm not an expert in objective-C, so what is actually missing?
- Could you give in words or a small drawing the call graph you expect?

**Comment by samuelchen2015** [comment]
"." in [self.eventHandler setup]  means calling self member "eventHander", then call method "setup"
I expect the call graph attach. 
![1724688419141](https://github.com/user-attachments/assets/28c0a183-f484-43ec-b19b-6b4f62398881)

**Comment by samuelchen2015** [comment]
to compare , it is ok for java
`// MainActivity.java
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View {

    private Presenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        presenter = new MainPresenter(this); 
        initView(); 
    }

    private void initView() {
        this.presenter.setup(); // call graph is fine for java
    }

}
`
![android](https://github.com/user-attachments/assets/a59063e0-7fdc-4b46-a392-181d54862d11)

**Comment by samuelchen2015** [comment]
this bug is still on latest doxygen version, hope fix it

**Comment by doxygen** [comment]
@samuelchen2015 Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by samuelchen2015** [comment]
> @samuelchen2015 Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

refer to the following fix, ti just fix only one ".", we shoud consider more "."。 for example：[self.viewmodel.eventhandler setup]。that means calling self ‘s’ proper Viewmodel ‘s proper Eventhander method setup。
`<ObjCCall>"self."{ID}             {`

**Comment by doxygen** [comment]
@samuelchen2015 You are right that the solution is rather specific to the problem reported. If you think the chaining of properties is something that is used in practice and needs to be supported, then please provide an example (I don't use Objective-C myself much, so I don't know what is and isn't used).

**Comment by samuelchen2015** [comment]
Attach my code. chaining of properties is always used. 
Hope to support it. thanks a lot
[code.zip](https://github.com/user-attachments/files/16829329/code.zip)


> @samuelchen2015 You are right that the solution is rather specific to the problem reported. If you think the chaining of properties is something that is used in practice and needs to be supported, then please provide an example (I don't use Objective-C myself much, so I don't know what is and isn't used).

**Comment by samuelchen2015** [comment]
> Attach my code. chaining of properties is always used. Hope to support it. thanks a lot [code.zip](https://github.com/user-attachments/files/16829329/code.zip)
> 
> > @samuelchen2015 You are right that the solution is rather specific to the problem reported. If you think the chaining of properties is something that is used in practice and needs to be supported, then please provide an example (I don't use Objective-C myself much, so I don't know what is and isn't used).

by the way , I am using 1.8.17 version and do a bit develop on it。 will you please tell how to fix it in  1.8.17？

**Comment by albert-github** [comment]
> by the way , I am using 1.8.17 version and do a bit develop on it。 will you please tell how to fix it in 1.8.17？

Why would you develop something based on the old version 1.8.17 and not based on the current (1.12.0) doxygen version?
So like I wrote before:
Doxygen version 1.8.17 is quite old (December 27, 2019) so I definitely would advise to update to the current 1.12.0 doxygen version.

**Comment by samuelchen2015** [comment]
> > by the way , I am using 1.8.17 version and do a bit develop on it。 will you please tell how to fix it in 1.8.17？
> 
> Why would you develop something based on the old version 1.8.17 and not based on the current (1.12.0) doxygen version? So like I wrote before: Doxygen version 1.8.17 is quite old (December 27, 2019) so I definitely would advise to update to the current 1.12.0 doxygen version.

Ok, I will try 1.12.0 . 
Just because code.l is hard to understand and transfering to update code take a bit effort. for example, how to understand the folloing code, what is use for 'startFontClass', 'endFontClass','codifyLines'
`            if (it!=yyextra->wordMap.end())
            {
              QCString word = it->second;
              bool isKeyword = word=="self";
              if (isKeyword)
              {
                startFontClass(yyscanner,"keyword");
              }
              codifyLines(yyscanner,word);
              if (isKeyword)
              {
                endFontClass(yyscanner);
              }
            }`

**Comment by albert-github** [comment]
It isn't easy and it needs some knowledge about (base and derived) classes in C++ and about how functions are called through the interfaces.
The `startFontClass` and `endFontClass` class are wrappers that again will call the relevant functions in the output generators (see e.g. htmlgen.cpp)

**Comment by doxygen** [comment]
@samuelchen2015 I've added a new commit to support chained properties. Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
